domain/annotation/state/state.go

```diff
@@ -99,12 +90,19 @@ WHERE c.name = $charmArgs.name AND c.revision = $charmArgs.revision;
 // This method is specialized to Models as opposed to the other Kinds because we
 // keep annotations per model, so we don't need to try to find the UUID of the
 // given ID (the model).
-func (st *State) getAnnotationsForModel(ctx context.Context, id annotations.ID, getAnnotationsStmt *sqlair.Statement) (map[string]string, error) {
+func (st *State) getAnnotationsForModel(ctx context.Context, id annotations.ID) (map[string]string, error) {
```

Comment:
Agreed, there are a few things in this domain that are weird. In this case I've just removed the parameter and annotated the errors with "model" rather than the ID.